File: javascript/leap/leap.js
// @ts-check

/**
 * Determines whether or not a given year is a leap year.
 *
 * @param {number} year
 * @returns {boolean} whether a year is a leap year
 */
export const isLeap = (year) => {
  const divisibleBy4 = year % 4 === 0
  const notDivisibleBy100 = year % 100 !== 0
  const divisibleBy400 = year % 400 === 0
  return divisibleBy400 || (divisibleBy4 && notDivisibleBy100)
};

File: erlang/leap/src/leap.erl
-module(leap).

-export([leap_year/1]).


leap_year(Year) when Year rem 400 == 0 -> true;
leap_year(Year) when (Year rem 4 == 0) and not(Year rem 100 == 0) -> true;
leap_year(_Year) -> false.
